### PR TITLE
[14.0][IMP] shopfloor qty picker

### PR DESCRIPTION
**Pull request by TDu**

### CODE REVIEW

**Comment by jbaudoux** [comment]
I'm testing it for non regression

File: shopfloor_mobile_base/static/wms/src/components/manual_select.js [hunk 1]
```diff
@@ -150,12 +150,13 @@ Vue.component("manual-select", {
...
         },
-        selected_color_klass(modifier) {
-            return (
-                "active " +
-                this.utils.colors.color_for("item_selected") +
-                (modifier ? " " + modifier : "")
-            );
+        selected_color_klass(rec, modifier) {
+            let color = this.utils.colors.color_for("item_selected");
+            if (rec.qty_done && rec.qty_done < rec.quantity)
```
Comment (simahawk) [inline shopfloor_mobile_base/static/wms/src/components/manual_select.js]: This should not stay here. ManualSelect is a generic component.
You should pass a new prop to support custom klass handler.

File: shopfloor_mobile/static/wms/src/wms_utils.js [hunk 2]
```diff
@@ -237,14 +239,15 @@ export class WMSUtils {
         });
         return options;
     }
-    move_line_qty_picker_options(line, override = {}) {
```
Comment (simahawk) [inline shopfloor_mobile/static/wms/src/wms_utils.js:240]: I would not remove this. I would just deprecate it.

**Review by simahawk — CHANGES_REQUESTED** [review]
For the rest, let's wait for our call on Mon ;)

File: shopfloor_mobile/static/wms/src/components/packaging-qty-picker.js [hunk 3]
```diff
@@ -1,28 +1,39 @@
 /**
  * Copyright 2020 Camptocamp SA (http://www.camptocamp.com)
  * @author Simone Orsi <simahawk@gmail.com>
- * Copyright 2021 Jacques-Etienne Baudoux (BCIM)
+ * Copyright 2021-2022 Jacques-Etienne Baudoux (BCIM)
```
Comment (simahawk) [inline shopfloor_mobile/static/wms/src/components/packaging-qty-picker.js]: do not set the end ;)

File: shopfloor_mobile_base/static/wms/src/components/manual_select.js (hunk 1 above)
Comment (jbaudoux) [reply shopfloor_mobile_base/static/wms/src/components/manual_select.js]: Moved as an extension of the component in shopfloor_mobile

File: shopfloor_mobile/static/wms/src/wms_utils.js (hunk 2 above)
Comment (jbaudoux) [reply shopfloor_mobile/static/wms/src/wms_utils.js:240]: done

File: shopfloor_mobile/static/wms/src/components/packaging-qty-picker.js (hunk 3 above)
Comment (jbaudoux) [reply shopfloor_mobile/static/wms/src/components/packaging-qty-picker.js]: ah!

**Comment by jbaudoux** [comment]
**Warning: This PR should not be merged standalone. All impacted scenario must be adapted.**

**Comment by simahawk** [comment]
> **Warning: This PR should not be merged standalone. All impacted scenario must be adapted.**

I don't get this... we should have only harmless changes here.

Please, note that the last commit is not for `shopfloor_mobile_base` is for `shopfloor_mobile`.

**Comment by simahawk** [comment]
#483 contains these changes

**Pull request closed without merging**